runtime/jcl/common/jcldefine.c

```diff
@@ -156,7 +156,8 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 	omrthread_monitor_enter(vm->classTableMutex);
 	/* Hidden class is never added into the hash table */
 	if ((NULL != utf8Name) && J9_ARE_NO_BITS_SET(*options, J9_FINDCLASS_FLAG_HIDDEN)) {
-		if (NULL != vmFuncs->hashClassTableAt(classLoader, utf8Name, utf8Length, 0)) {
+		J9Class *frozenJCLClazz = vmFuncs->hashClassTableAt(classLoader, utf8Name, utf8Length, J9_FINDCLASS_FLAG_DONT_SKIP_FROZEN_JCL_DEFINE_CLASS);
```

Comment:
This name is misleading: the class isn't necessarily frozen; perhaps `persistedClazz` (see line 164)?